core/domain/prod_validation_jobs_one_off.py

```diff
@@ -648,13 +747,38 @@ def _validate_explorations_are_public(cls, item):
                     item.id, private_exp_ids))
 
     @classmethod
-    def _validate_collections_are_public(cls, item):
+    def _validate_collections_are_public(
+            cls, item, field_name_to_external_model_references):
         """Validates that collections for model are public.
 
         Args:
             item: ndb.Model. BaseUserModel to validate.
+            field_name_to_external_model_references:
+                dict(str, (list(ExternalModelReference))).
+                A dict is keyed by field name. Each value consists of a list
+                of ExternalModelReference objects.
         """
-        col_ids = cls._get_col_ids(item)
+        if 'collection_ids' not in field_name_to_external_model_references:
+            return
+
+        col_ids = []
+        collection_model_references = (
+            field_name_to_external_model_references['collection_ids'])
+
+        for collection_model_reference in collection_model_references:
+            model_class = collection_model_reference.class_name
+            model_id = collection_model_reference.model_id
```

Comment:
For these two properties, declare them only when you need them (i.e. in the "if").

Similar comments apply elsewhere.

Reply:
Done.